wolfcrypt/src/asn.c

```diff
@@ -379,7 +364,7 @@ struct tm* mqx_gmtime(const time_t* clock, struct tm* tmpTime)
 
 time_t ksdk_time(time_t* timer)
 {
-    time_t localTime;
+    static time_t localTime;
```

Comment:
A static time_t type here loses thread safety.  If the caller pass in a NULL pointer for timer we're not assigning the user pointer to localTime (just a copy, the user passed in NULL and can't "retreive" it after the call).  We're just using the timer pointer as a shortcut in case it is a valid pointer so we don't have to duplicate the copy and return.